src/gluonts/nursery/glide/pipeline.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import multiprocessing as mp
+from typing import Callable, Iterator, TypeVar, Generic
+
+from toolz.itertoolz import concat
+from toolz.itertoolz import partition_all as batcher
+
+from ._partition import partition
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+class sentinel:
+    pass
+
+
+def map_to_queue(fn, emitter, queue, batch_size):
+    stream = map(fn, emitter)
+    for batch in batcher(batch_size, stream):
+        queue.put(batch)
+    queue.put(sentinel)
+
+
+class Map:
+    def __init__(self, fn, partitions: list):
+        self.fn = fn
+        self.partitions = partitions
+
+    def __iter__(self):
+        yield from concat(
+            map(self.fn, partition) for partition in self.partitions
+        )
+
+
+class ParPipelineIterator:
```

Comment:
I don’t get the name “pipeline” (neither for this nor for the module): an object of this class is what you get when you call `iter` on a `ParMap` object. Shouldn’t it be `ParMapIterator`?

Reply:
Yes, I used to call `ParMap` `Pipeline` but forgot to also rename this :(